src/types.jl

```diff
@@ -1,143 +1,219 @@
+end
+
+function ImageClassifier(
+    ; builder::B = image_builder(VGGHack),
+    finaliser::F = Flux.softmax,
+    optimiser::O = Optimisers.Adam(),
+    loss::L = Flux.crossentropy,
+    epochs = 10,
+    batch_size = 1,
+    lambda = 0,
+    alpha = 0,
+    rng = Random.default_rng(),
+    optimiser_changes_trigger_retraining = false,
+    acceleration = CPU1(),
+) where {B, F, O, L}
+    model = ImageClassifier{B, F, O, L}(
+        builder,
+        finaliser,
+        optimiser,
+        loss,
+        epochs,
+        batch_size,
+        lambda,
+        alpha,
+        rng,
+        optimiser_changes_trigger_retraining,
+        acceleration,
+    )
+
+    message = clean!(model)
+    isempty(message) || @warn message
+
+    return model
+end
+
 
 MMI.metadata_pkg.(
-  (
-      NeuralNetworkRegressor,
-      MultitargetNeuralNetworkRegressor,
-      NeuralNetworkClassifier,
-      ImageClassifier,
-      NeuralNetworkBinaryClassifier,
-  ),
-    name="MLJFlux",
-    uuid="094fc8d1-fd35-5302-93ea-dabda2abf845",
-    url="https://github.com/alan-turing-institute/MLJFlux.jl",
-    julia=true,
-    license="MIT",
+    (
+        NeuralNetworkRegressor,
+        MultitargetNeuralNetworkRegressor,
+        NeuralNetworkClassifier,
+        ImageClassifier,
+        NeuralNetworkBinaryClassifier,
+    ),
+    name = "MLJFlux",
+    uuid = "094fc8d1-fd35-5302-93ea-dabda2abf845",
+    url = "https://github.com/alan-turing-institute/MLJFlux.jl",
+    julia = true,
+    license = "MIT",
 )
 
+const MODELSUPPORTDOC = """
+This model supports categorical columns in the input table. If present, such columns are embedded into
+dense vectors by the use of an additional `EntityEmbedder` layer after the input 
+as described in Entity Embeddings of Categorical Variables by Cheng Guo, Felix Berkhahn.
+"""
+
+const XDOC = """
+- `X` is either a `Matrix` or any table of input features (eg, a `DataFrame`) whose columns are of scitype
+  `Finite`; check column scitypes with `schema(X)`. If `X` is a `Matrix`, it is assumed to have columns corresponding to features and rows corresponding to observations. 
+  In case any `Multiclass` or `OrderedFactor` columns appear in `X`, the constructed network will use an `EntityEmbedder` layer to transform them into dense vectors.
+"""
 
+const EMBDOC = """
+- `embedding_dims` is a `Dict` whose keys are column names of categorical columns, given as symbols, and whose values are real numbers representing the desired dimensionality
+  of the entity embeddings of such columns. An integer value such as `7` would set the embedding dimensionality of such columns to `7`. Meanwhile, a float value such as `0.5` 
+  would set the embedding dimensionality of such columns to `ceil(0.5 * number of levels in column)`. Any unspecified columns will by default have their values set to either 0.5
+  or 0.2 depending on whether the number of levels in the column is less than 20 or greater than 20 respectively.
+"""
```

Comment:
I still think the second part of this default (the complication with 20) makes this overly complicated for a heuristic for which we have no solid theoretical/empirical justification. I'd just make the default 0.5 and be done with it. For large cardinality, the user needs to step in and use some use-case specific reasoning. They're often going to cap this depending on time and memory constraints, which we cannot anticipate here, right?

Reply:
OK. I will first revisit their paper to see if they suggested something and if there is nothing I think I will do `10`. I would rather let the user figure out that they have to override the default to get more optimal results than have them override it after a few minutes of frustration due to an unexpectedly large network. What do you think @ablaom ?

Reply:
👍🏾

Reply:
Done in daa4b2aed5ce4d345c92182413583d3667a660bc